modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/APIService.java

```diff
@@ -260,4 +262,89 @@ public Response importAPI(@Multipart("file") InputStream uploadedInputStream, @Q
+     */
+    @PUT
+    @Path("/{apiID}")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateAPI(@Multipart("file") InputStream uploadedInputStream, @PathParam("apiID") String apiID,
+                              @QueryParam("preserveProvider") String defaultProviderStatus, @Context HttpHeaders httpHeaders) {
+        boolean isTenantFlowStarted = false;
+        boolean isProviderPreserved = true;
+
+        //Check if the URL parameter value is specified, otherwise the default value "true" is used
+        if (APIImportExportConstants.STATUS_FALSE.equalsIgnoreCase(defaultProviderStatus)) {
+            isProviderPreserved = false;
+        }
+
+        try {
+            Response authorizationResponse = AuthenticatorUtil.authorizeUser(httpHeaders);
+
+            if (Response.Status.OK.getStatusCode() != authorizationResponse.getStatus()) {
+                return authorizationResponse;
+            }
+
+            AuthenticationContext authenticationContext = (AuthenticationContext) authorizationResponse.getEntity();
+            String tenantDomain = MultitenantUtils.getTenantDomain(authenticationContext.getUsername());
+            String currentUser = authenticationContext.getDomainAwareUsername();
+
+            if (!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+                currentUser = currentUser + "@" + tenantDomain;
+            }
+            APIImportUtil.initializeProvider(currentUser);
+
+            //Temporary directory is used to create the required folders
+            String currentDirectory = System.getProperty(APIImportExportConstants.TEMP_DIR);
+            String createdFolders = File.separator +
+                    RandomStringUtils.randomAlphanumeric(APIImportExportConstants.TEMP_FILENAME_LENGTH) +
+                    File.separator;
+            File importFolder = new File(currentDirectory + createdFolders);
+            boolean folderCreateStatus = importFolder.mkdirs();
+
+            //API import process starts only if the required folder is created successfully
+            if (folderCreateStatus) {
+                String uploadFileName = APIImportExportConstants.UPLOAD_FILE_NAME;
+                String absolutePath = currentDirectory + createdFolders;
+                APIImportUtil.transferFile(uploadedInputStream, uploadFileName, absolutePath);
+
+                String extractedFolderName;
+                try {
+                    extractedFolderName = APIImportUtil.extractArchive(
+                            new File(absolutePath + uploadFileName), absolutePath);
+                } catch (APIImportException e) {
+                    return Response.status(Status.BAD_REQUEST).entity(e.getMessage()).build();
+                }
+
+                if (tenantDomain != null &&
+                        !org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME
```

Comment:
Why are we having the package name here? we can use this right MultitenantConstants.SUPER_TENANT_DOMAIN_NAME ?

Reply:
yeah will refactor